tests/testGroupScoreAndSync.js

```diff
@@ -0,0 +1,355 @@
+var ERR = require('async-stacktrace');
+
+const config = require('../lib/config');
+const _ = require('lodash');
+const path = require('path');
+var assert = require('chai').assert;
+var request = require('request');
+var cheerio = require('cheerio');
+
+var sqldb = require('@prairielearn/prairielib').sqldb;
+var sqlLoader = require('@prairielearn/prairielib').sqlLoader;
+var sql = sqlLoader.loadSqlEquiv(__filename);
+var page, form, elemList; 
+const helperServer = require('./helperServer');
+const courseDir = path.join(__dirname, '..', 'testCourse');
+
+const locals = {};
+locals.helperClient = require('./helperClient');
+locals.siteUrl = 'http://localhost:' + config.serverPort;
+locals.baseUrl = locals.siteUrl + '/pl';
+locals.courseInstanceUrl = locals.baseUrl + '/course_instance/1';
+locals.courseInstanceBaseUrl = locals.baseUrl + '/course_instance/1';
+
+const storedConfig = {};
+const question = [{qid: 'addNumbers', type: 'Freeform', maxPoints:5}];
+const questions = _.keyBy(question, 'qid');
+
+describe('assessment instance group synchorization test', function () {
+    this.timeout(10000);
+
+    before('set authenticated user', function(callback) {
+        storedConfig.authUid = config.authUid;
+        storedConfig.authName = config.authName;
+        storedConfig.authUin = config.authUin;
+        callback(null);
+    });
+    before('set up testing server', helperServer.before(courseDir));
+    after('shut down testing server', helperServer.after);
+    describe('1. database initialization', function() {
+        it('get group-based homework assessment id', function(callback) {
+            sqldb.query(sql.select_group_work_assessment, [], function(err, result) {
+                if (ERR(err, callback)) return;
+                assert.notEqual(result.rowCount, 0);
+                assert.notEqual(result.rows[0].id, undefined);
+                locals.assessment_id = result.rows[0].id;
+                locals.assessmentUrl = locals.courseInstanceBaseUrl + '/assessment/' + locals.assessment_id;
+                locals.instructorAssessmentsUrlGroupTab = locals.courseInstanceBaseUrl + '/instructor/assessment/' + result.rows[0].id + '/groups';
+                locals.questionBaseUrl = locals.courseInstanceBaseUrl + '/instance_question';
+                locals.assessmentsUrl = locals.courseInstanceBaseUrl + '/assessments';
+                callback(null);
+            });
+        });
+        
+    
+        
+    });
+    describe('2. GET to instructor assessments URL group tab for the first assessment', function() {
+        it('should load successfully', function(callback) {
+            request(locals.instructorAssessmentsUrlGroupTab, function(error, response, body) {
+                if (error) {
```

Comment:
Should we be using the `if (ERR(error, callback)) return;` convention here?

Reply:
Yes. Fix all.